### ISSUE TITLE
Unable to run `next` with custom servers.

### ISSUE THREAD

**Issue by AlphaNecron** [vercel/next.js#41974]
### Verify canary release

- [x] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```
    Operating System:
      Platform: linux
      Arch: x64
      Version: #1 SMP Sat, 22 Oct 2022 13:04:31 +0000
    Binaries:
      Node: 16.16.0
      npm: 8.19.2
      Yarn: 3.2.4
      pnpm: N/A
    Relevant packages:
      next: 13.0.1-canary.0
      eslint-config-next: 13.0.0
      react: 18.2.0
      react-dom: 18.2.0
```

### What browser are you using? (if relevant)

Firefox Developer Edition 107.0b4-1

### How are you deploying your application? (if relevant)

TypeScript runner

#### Bug 1

Always getting this when starting `next` with custom server.
```js
error - Failed to load next.config.js, see more info here https://nextjs.org/docs/messages/next-config-error
/data/Dev/Void/node_modules/next/dist/server/config.js:70
                userConfigModule = await import((0, _url).pathToFileURL(path).href);
                                                    ^

TypeError: (0 , _url).then is not a function
    at Object.loadConfig (/data/Dev/Void/node_modules/next/dist/server/config.js:70:53)
    at NextServer.prepare (/data/Dev/Void/node_modules/next/dist/server/next.js:131:24)
```

#### Bug 2

I also get this when opening `dev` server:
![image](https://user-images.githubusercontent.com/57827456/198480575-bec336b4-22dc-4e58-832e-43908fb2720b.png)
It points to this line:
![image](https://user-images.githubusercontent.com/57827456/198480911-09352cca-35db-472f-88a1-ea9ef5c4bbc4.png)

#### Bug 1

Run the server with `tsx` or `ts-node` and it will crash with above error. Tested with both `ts-node` and `tsx`.

**Workaround:** `export  __NEXT_TEST_MODE=jest`, which forces `next` to import with `require()` instead of dynamic import.

#### Bug 2
An error caused by `webpack` is thrown to the console:
`Uncaught ReferenceError: __name is not defined`

### Link to reproduction

https://github.com/AlphaNecron/next-issue

#### Bug 1

Run the server with `start` script and it will crash with above error. Tested with both `ts-node` and `tsx`.

#### Bug 2
1. Use the workaround aforementioned.
2. Build with `next build`
3. Start the server with `yarn start` (It works flawlessly)
4. Close the server and execute `dev` script to start the `dev` server.
5. Open `localhost:3000` and check out the console.
6. Launch the production server again  with `yarn start`.
7. The production build should be corrupted and it will throw:
```Error: Could not find a production build in the 'reproduction-template/.next' directory. Try building your app with 'next build' before starting the production server.```

**Comment by AlphaNecron** [comment]
Added another bug, bump

**Comment by AlphaNecron** [comment]
Are there other ways to use Next.JS with custom TypeScript servers?

**Comment by AlphaNecron** [comment]
Fixed by downgrading `tsx` to `3.9.0`, `esbuild>=0.15` is not backward-compatible therefore breaking everything.
